src/Tribe/Commerce/PayPal/Main.php

```diff
@@ -546,42 +546,38 @@ public function attendance_totals() {
 	 *
 	 * @since 4.7
 	 *
-	 * @param array  $data
-	 * @param string $order_id
-	 * @param int    $event_id
+	 * @param array $attendee_data Information that we are trying to save.
+	 * @param int   $attendee_id   The attendee ID.
+	 * @param int   $post_id       The event/post ID.
 	 */
-	public function update_attendee_data( $data, $order_id, $event_id ) {
+	public function update_attendee_data( $attendee_data, $attendee_id, $post_id ) {
 		$user_id = get_current_user_id();
 
-		$ticket_orders    = $this->tickets_view->get_post_ticket_attendees( $event_id, $user_id );
-		$ticket_order_ids = wp_list_pluck( $ticket_orders, 'order_id' );
+		$ticket_attendees    = $this->tickets_view->get_post_ticket_attendees( $post_id, $user_id );
+		$ticket_attendee_ids = wp_list_pluck( $ticket_attendees, 'attendee_id' );
 
-		// This makes sure we don't save attendees for orders that are not from this current user and event
-		if ( ! in_array( $order_id, $ticket_order_ids ) ) {
+		// This makes sure we don't save attendees for attendees that are not from this current user and event.
+		if ( ! in_array( $attendee_id, $ticket_attendee_ids, true ) ) {
 			return;
 		}
 
-		// Get the Attendee Data, it's important for testing
-		foreach ( $ticket_orders as $test_attendee ) {
-			if ( $order_id !== $test_attendee['order_id'] ) {
-				continue;
-			}
-		}
-
-		$attendee_email     = empty( $data['email'] ) ? null : sanitize_email( $data['email'] );
+		$attendee_email     = empty( $attendee_data['email'] ) ? null : sanitize_email( $attendee_data['email'] );
 		$attendee_email     = is_email( $attendee_email ) ? $attendee_email : null;
-		$attendee_full_name = empty( $data['full_name'] ) ? null : sanitize_text_field( $data['full_name'] );
-		$attendee_optout    = empty( $data['optout'] ) ? 0 : max( 0, (int) $data['optout'] );
+		$attendee_full_name = empty( $attendee_data['full_name'] ) ? null : sanitize_text_field( $attendee_data['full_name'] );
+		$attendee_optout    = empty( $attendee_data['optout'] ) ? 0 : (int) tribe_is_truthy( $attendee_data['optout'] );
 
-		update_post_meta( $order_id, $this->attendee_optout_key, (int) $attendee_optout );
+		$attendee_data_to_save = [
+			'full_name' => $attendee_full_name,
+			'email'     => $attendee_email,
+			'optout'    => $attendee_optout,
+		];
 
-		if ( ! is_null( $attendee_full_name ) ) {
-			update_post_meta( $order_id, $this->full_name, $attendee_full_name );
-		}
+		// Remove arguments that are null.
+		$attendee_data_to_save = array_filter( $attendee_data_to_save, static function ( $value ) {
```

Comment:
* :no_entry_sign: Opening parenthesis of a multi-line function call must be the last content on the line
* :no_entry_sign: Only one argument is allowed per line in a multi-line function call

<sub>hash: 0f802a557732477c41fc55ad7657e207</sub>